### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by pablomendezroyo**
## Context
- **Update https-ipfs-client dependency**
- **Update node engine version**: in Dockerfile and gha
- **Use of local ipfs node in int tests**
- **Get dappnode ipfs gateway id dynamically**
Int tests eventually need to startup a local IPFS node for testing, this local IPFS node needs to connect to the dappnode IPFS gateway. To accomplish that, it is required the node ID, which was been obtained statically which cause errors when this ID changes

## Approach

Get dappnode ipfs gateway ID dynamically

## Test instructions

- Ensure int tests works as expected with this change

### CODE REVIEW

File: packages/dappmanager/test/modules/ipfs/index.test.int.ts [hunk 1]
```diff
@@ -31,11 +26,17 @@ describe("ipfs / integration test", function () {
   });
 
   it("Upload directory", async () => {
-    dirHash = await ipfsAddDirFromFs(dirPath);
+    const addResults = await ipfsAddAll(dirPath);
+    dirHash =
+      addResults
+        .find(addResult => addResult.path === "test_files/ipfs-test-upload")
+        ?.cid.toString() || "";
+    if (!dirHash) throw Error("No directory hash found");
```
Comment (dapplion) [inline packages/dappmanager/test/modules/ipfs/index.test.int.ts:34]: Abstract this into a separate function, which takes `dirPath` as argument and returns you the CID of the root directory

File: packages/dappmanager/test/integrationSpecs/buildReleaseDirectory.ts [hunk 2]
```diff
@@ -71,7 +71,9 @@ export async function uploadDirectoryRelease({
   if (setupWizard) writeJson("setup-wizard.json", setupWizard);
   if (disclaimer) writeAsset("disclaimer.md", disclaimer);
 
-  const rootHash = await ipfsAddDirFromFs(releaseDir);
+  const addResults = await ipfsAddAll(releaseDir);
+  // The last result is the root /test_files, the second is the dir /test_files/something/
+  const rootHash = addResults[addResults.length - 2].cid.toString();
```
Comment (dapplion) [inline packages/dappmanager/test/integrationSpecs/buildReleaseDirectory.ts:76]: Re-use function from https://github.com/dappnode/DNP_DAPPMANAGER/pull/1018/files#r868979902

File: packages/dappmanager/test/modules/ipfs/index.test.int.ts (hunk 1 above)
Comment (pablomendezroyo) [reply packages/dappmanager/test/modules/ipfs/index.test.int.ts:34]: the returns from the `ipfsAdd` and the new `ipfsAddAll` have changed so the function `ipfsAddDirFromFs` did not work as expected

**Review by dapplion — APPROVED** [review]
LGTM! :fire:

**Pull request merged**